app/screens/home/index.tsx

```diff
@@ -39,6 +42,14 @@ type HomeProps = LaunchProps & {
 
 const Tab = createBottomTabNavigator();
 
+// This is needed since the Database Provider is recreating this component
+// when the database is changed (couldn't find exactly why), re-triggering
```

Comment:
When switching the active database, the provider changes the context and thus triggering a re-render, I guess that is causing this component to unmount and then mount again

Reply:
The thing is that in normal scenarios, a re-render should not unmount and mount again, it should just change the props. But didn't explore deep enough to find the real culprit here.